blocks/components/editable/index.js

```diff
@@ -101,14 +110,26 @@ export default class Editable extends wp.element.Component {
 		this.props.onFocus();
 	}
 
-	onChange() {
-		if ( ! this.editor.isDirty() ) {
+	isActive() {
+		return tinymce.EditorManager.activeEditor.id === this.editor.id;
+	}
+
+	onSelectionChange() {
+		// We must check this because selectionChange is a global event.
+		if ( ! this.isActive() ) {
 			return;
 		}
 
-		this.savedContent = this.getContent();
-		this.editor.save();
-		this.props.onChange( this.savedContent );
+		this.content = this.getContent();
+		this.selection = this.getSelection();
+
+		if ( ! this.props.onChange ) {
+			return;
+		}
+
+		if ( ! isEqual( this.props.value, this.content ) ) {
```

Comment:
>@aduth What would you suggest? Calling onChange all the time? We might want to do it anyway later if we set the selection too.

Personally I didn't see it being hugely problematic that we only triggered the change callback when removing focus from the block. Another issue we have with this is that `UPDATE_BLOCK` is repeatedly dispatched while typing. This might be ideal from the perspective of ensuring the Editable value and state are kept in sync, but is fairly wasteful to reconcile. For example, with these changes we're not only doing a deep equality here, but another needless one in `componentDidUpdate`.